### DISCUSSION TITLE
HW interface to make it clear students won't lose marks

### DISCUSSION THREAD

**Discussion by firasm** [PrairieLearn/PrairieLearn#6658] [Ideas]
I posted this on Slack, but students are often confused about whether or not they can generate new variants to try the problem again. They're afraid (reasonably) they'll lose marks if they try questions again and aren't able to get it right (which sort of defeats the purpose of designing algorithmically generated problems).

> “If I redo questions and reinput my answers will that change my mark to 50%? Right now I’m scared to go back through the homework questions and try different variants since I’m scared I’ll accidentally alter the grade I earned originally.”

Is there something that others do (other than repeat it again and again) to help deal with this?

Brainstorm ideas below!

**Comment by firasm** [comment]
## Idea 1: Add a small "?" beside the score with some information

<img width="1325" alt="two options" src="https://user-images.githubusercontent.com/2507459/203613969-b2474d36-2aae-4f44-820e-a2e9341b6039.png">

**Comment by firasm** [comment]
## Idea 2: Change the colour/style of the buttons

On subsequent variants (after they got 100%) make it clearer their score won't change by changing the buttons.

<img width="957" alt="Screenshot 2022-11-23 at 9 46 57 AM" src="https://user-images.githubusercontent.com/2507459/203615145-bdff1a8e-b678-4f7a-8b7c-bc5fe98b86ee.png">

This is probably a LOT of extra work.

**Comment by firasm** [comment]
## Idea 3: Revamp the "Available Credit" column to show more useful information

Currently this column is populated by the access rule, but in my context this becomes confusing because students can get up to 50% even until the end of the course..

<img width="1306" alt="Screenshot 2022-11-23 at 9 30 32 AM" src="https://user-images.githubusercontent.com/2507459/203615411-31cc6eca-3296-47fe-bba7-dee746b8cb78.png">

**Reply by mwest1066** [reply]
Excellent point! Does the help text for this clearly say that your score can't decrease? That would be an easy fix, although of course it does assume that people click the help text button.

**Reply by firasm** [reply]
The help text just has the list of Available credit rules (complicated for my course)

<img width="347" alt="Screenshot 2022-11-23 at 11 08 29 AM" src="https://user-images.githubusercontent.com/2507459/203628293-09dd19fc-bd0b-44e3-a8ad-649c4317db50.png">

**Comment by ffund** [comment]
I actually think it should go here:

![image](https://user-images.githubusercontent.com/2316553/203615860-73ea2caa-b45a-4861-b90e-a044401c6b3c.png)

On exams when there are multiple attempts, students have learned to look at that text to see what the implications of submitting again would be. So it makes sense that for homework, the implications of resubmission should be in the same place.

**Reply by firasm** [reply]
Oh this is a good place too!

**Comment by firasm** [comment]
## Idea 5: Use the "text" property to add a banner-type thing on the assessment page

Thanks to @echuber2 for pointing out this one, TIL there's a text property I can add to all my assignments!

<img width="1349" alt="Screenshot 2022-11-23 at 11 15 01 AM" src="https://user-images.githubusercontent.com/2507459/203629130-d2eba33f-d236-481f-8ae5-d9299ae37eca.png">
